bootstrap/kubeadm/controllers/alias.go

```diff
@@ -42,6 +43,9 @@ type KubeadmConfigReconciler struct {
 	// WatchFilterValue is the label value used to filter events prior to reconciliation.
 	WatchFilterValue string
 
+	// Filter checks if a KubeadmConfig object should be reconciled.
+	Filter predicates.Filter
```

Comment:
/hold

My personal opinion is that we should not add a new knob to filter issues we reconcile which is orthogonal to WatchFilterValue and its possible improvements being discussed in https://github.com/kubernetes-sigs/cluster-api/issues/7775. 
IMO those efforts should converge and come up with a common definition of what a filter is, that could translate into e.g. a watch filter struct or other solutions.

Also, worth to notice that such type of change in order to actually work should be accepted by provider implementers as well, and this requires a discussion during office hours or a small proposal to rally on async.

Reply:
I am ok with moving discussion to  #7775.